[2018.3] Merge forward from 2017.7 to 2018.3

rallytime opened the pull request:
Conflicts:
  - salt/modules/file.py
  - salt/modules/win_service.py
  - salt/states/mount.py
  - salt/states/service.py
  - tests/integration/__init__.py
  - tests/integration/runners/test_state.py
  - tests/integration/states/test_file.py
  - tests/unit/states/test_file.py

Commit:
Fixed partition names with spaces effectively containing only the first word.

zer0def committed:
Merge branch '2017.7' into parted-naming

dwoz committed:
Install the launcher so we can execute py files

dwoz committed:
Merge pull request #49103 from dwoz/cmd_test_fix

Install the launcher so we can execute py files

dwoz committed:
Multiple fixes for integration.states.test_file

- Fix wart in PR #49087 file://c:\foo\bar vs c:\foo\bar local paths
- Finalize fix for test_issue_8343_accumulated_require_in
- Fix wart in PR #49088 Ignore proper directory test

Commit:
Merge pull request #49118 from dwoz/test_file_fixes

Multiple fixes for integration.states.test_file

ralish committed:
Remove obsolete documentation on external_nodes setting

This setting is no longer used anywhere in the codebase having been
superseded by the master_tops setting many releases ago.

Ch3LL committed:
[2017.7] Update bootstrap script to latest release (2018.08.15)

gtmanfred committed:
Merge pull request #49136 from Ch3LL/bootstrap_2017.7

[2017.7] Update bootstrap script to latest release (2018.08.15)

Ch3LL committed:
Remove -Z script_arg for cloud tests

twangboy committed:
Add timeout parameter

Add a timeout parameter to start, stop, restart, and delete

twangboy committed:
Add timeout support to the state

Remove redundent net stop command from the stop function
Remove extra _enable call on Windows that was causing incorrect state
return comments

damon-atkins committed:
2017_win_service_damon

Merge this if you like the suggestion.

twangboy committed:
Merge pull request #10 from damon-atkins/patch-1

2017_win_service

erwindon committed:
Fixed unknown 'exceptions' under Python3 (#49152)

dwoz committed:
Allow test suite to finish if tmp dir removal fails

dwoz committed:
add Exception type

terminalmage committed:
Fix bug in keep_source for non-templated salt:// file sources

Unless the file is one of A) a remote file source (http/https/ftp/etc.)
or B) being templated, then `file.get_managed` does not cache the file.
Thus, the variable we rely on to tell us the cached path is an empty
string, and we don't remove it even though the file would eventually
later be cached when we run `file.manage_file`. This adds some
additional logic to ensure that we do check if the file has been cached
if `sfn` is an empty string.

dwoz committed:
Merge pull request #49163 from dwoz/tmp_dir

Allow test suite to finish if tmp dir removal fails

terminalmage committed:
Add keep_source integration tests

twangboy committed:
Fix start/stop functions

Fix typo in the dict key 'Status'
Standardize docs
Fix some lint issues
Rename the new helper function to _status_wait
Add urls to code comments
Add version added statements

twangboy committed:
Remove testing debug stuff

twangboy committed:
Improve timeout in delete

Improved error handling

twangboy committed:
Fix some lint (remove whitespace)`

dwoz committed:
Fix remaining file state integration tests (py3)

dwoz committed:
Simplify dict keys lookup

erwindon committed:
Improved solution as per @dwoz's suggestion

erwindon committed:
Must have 2 lines between imports and code

gtmanfred committed:
Merge pull request #49167 from dwoz/filefix

Fix remaining file state integration tests (py3)

Commit:
Merge branch '2017.7' into issue49154

Commit:
Merge branch '2017.7' into wheel_error_error

Commit:
Merge pull request #49143 from Ch3LL/bp-49142

[2017.7] Backport #49142

dwoz committed:
Merge pull request #49162 from erwindon/wheel_error_error

Fixed unknown 'exceptions' under Python3 (#49152)

dwoz committed:
Account for normalized dirs in unit tests

dwoz committed:
Fix directory unit test

ralish committed:
Update documentation to correctly state enable_gpu_grains default

While the relevant code in grains/core.py does enable GPU grains by
default, the default in config/__init__.py for the DEFAULT_MASTER_OPTS
dictionary is disabled. This takes precedence over the default in
grains/core.py which would only take effect if the enable_gpu_grains
setting was not specified at all.

dwoz committed:
Merge pull request #49179 from dwoz/test_unit_states_file

Account for normalized dirs in unit tests

erwindon committed:
Fix for #45620: "Salt CLI is rounding floats to 2 decimal places" (actually: Salt CLI is using only 12 digits for precision)

use repr() to get the full precision also for older python versions
as until about python32 it was limited to 12 digits only by default

erwindon committed:
trailing whitespace removal

Commit:
Merge branch '2017.7' into external_nodes_docs

Commit:
Merge pull request #49164 from terminalmage/issue49154

Fix bug in keep_source for non-templated salt:// file sources

Commit:
Merge pull request #48032 from zer0def/parted-naming

Fixed partition names with spaces effectively containing only the first word.

Commit:
Merge pull request #49185 from ralish/enable_gpus_grains_doc

Update documentation to correctly state enable_gpu_grains default

Commit:
Merge pull request #49184 from ralish/external_nodes_docs

Remove obsolete documentation on external_nodes setting

gtmanfred committed:
Merge pull request #49187 from erwindon/py27fixnumberformat

Fix for #45620: "Salt CLI is rounding floats to 2 decimal places"

twangboy committed:
Work with seconds

minusf committed:
check for mandatory parameters to avoid false positives

minusf committed:
fix some underhanging indent while here...

twangboy committed:
Use os.linesep.join instead of textwrap.dedent

terminalmage committed:
Allow compound matching in eauth config expressions

When we try to match the configured expression, we look for a match type
(e.g. `I@`, `G@`, etc.) at the beginning, and when we don't find one we
were falling back to a glob. This changes the fallback to compound so
that we can support compound matches. Note that the compound matching
engine will act just like the glob match engine when the host passed to
it is a single minion ID glob, so using compound as the fallback gets
you glob matching for free.

dwoz committed:
Revert "Multiple block replace test fixes"

This reverts commit 94ac2b4fc70d6b2b971c6aafa1f82ea11c983e4e.

dwoz committed:
Better blockfix replace

dwoz committed:
Use six to make sure content is unicode

gtmanfred committed:
Merge pull request #49236 from terminalmage/issue32737

Allow compound matching in eauth config expressions

Commit:
Merge pull request #49242 from dwoz/blockreplace_better_fix

Blockreplace better fix

Commit:
Merge pull request #49231 from minusf/mount-check-name

Check for mandatory parameters for `mount.mounted`

gtmanfred committed:
mark orchestration state tests as flaky

gtmanfred committed:
flaky tests are flaky yo

Commit:
Merge pull request #49259 from gtmanfred/flaky-2017.7

[2017.7] flaky tests

mattp- committed:
extend #488588 to cover SyncAuth class

the original fix was only covered asyncauth, but the syncauth method overrode
the fix. This patch duplicates it so behavior should be consistent in both
codepaths.

gtmanfred committed:
Merge pull request #49144 from twangboy/fix_48163

Add timeout parameter to win_service functions

gtmanfred committed:
Merge pull request #49262 from bloomberg/short-circuit

extend #48588 to cover SyncAuth class

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - salt/modules/file.py
  - salt/modules/win_service.py
  - salt/states/mount.py
  - salt/states/service.py
  - tests/integration/__init__.py
  - tests/integration/runners/test_state.py
  - tests/integration/states/test_file.py
  - tests/unit/states/test_file.py

Commit:
Update old utils paths with new paths

dwoz commented on tests/unit/states/test_file.py:
```diff
@@ -870,7 +870,7 @@ def test_directory(self):
                             'result': None,
                             'pchanges': p_chg,
-                            'changes': {}
+                            'changes': {name: {'directory': 'new'}}
```
I think we just want to skip this change.

https://jenkinsci.saltstack.com/job/pr-kitchen-ubuntu1604-py2/job/PR-49282/1/testReport/junit/unit.states.test_file/TestFileState/test_directory/

Commit:
Fix unit test for file state

Chose the wrong side of a merge conflict resolution. This commit restores
the correct testing behavior.

Commit:
Call the `with_tempfile` decorator correctly for 2018.3

rallytime commented:
The last 2 commits should fix 2 of the test failures. I need to investigate the final test failure more and discuss a fix from @terminalmage in the morning.

Commit:
Add 'minion_data_cache: True' to mocked opts for minions unit tests

rallytime commented:
OK, this should be good now, pending tests.
